irohad/model/execution/domain_name_parser.hpp

```diff
@@ -0,0 +1,38 @@
+#ifndef IROHA_DOMAIN_NAME_PARSER_HPP
+#define IROHA_DOMAIN_NAME_PARSER_HPP
+
+#include <string>
+
+namespace iroha {
+  namespace model {
+    /**
+     * Parse a string as a domain name as defined in RFC 1035.
+     */
+    class DomainNameParser {
```

Comment:
I think there's no need in class.
Also I don't really sure that *irohad/model/execution* is best folder for it, at least for me it's not obvious. Maybe *libs/parser/*?

Reply:
`libs/util`

Reply:
I can create a package `libs/util` and move the `domain_name_parser` there. What should I name the package? `util` sounds like too generic and might conflict with other packages. Do you have any suggestion?

Reply:
`validator` is fine. All it does is answers the question "is domain name valid" and imo `validator` sounds okay.